### PR TITLE
Bug 1680019 - Add weather endpoint

### PR DESCRIPTION
**Pull request by htwyford**
We use two different weather endpoints in the extension, so we need two different endpoints on the server. An early version of this patch used a single endpoint (/cid/weather) and passed the pathname of the specific endpoint in an X-Path header. The problem with this is that the client only saw repeated requests to `https://topsites.mozilla.com/cid/weather`, so it would cache data far too aggressively. We need to actually call different paths using different query parameters to get appropriate caching behaviour.

As alluded to above, this change also adds support for clients to pass query paramters in their requests to the server. These query paramters are then merged with the params pre-defined in CONFIG. For example, consider a request to `https://topsites.mozilla.com/cid/weather_location?q=toronto`. The `q` parameter is merged with the `apikey` parameter predefined in CONFIG to create a final URL of
`https://<WEATHER_LOCATION_URL>?q=toronto&apikey=abc`

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -47,49 +50,90 @@ const CONFIG = {
       sub3: `${SPECIAL_DELIM}header${SPECIAL_SEP}x-source${SPECIAL_DELIM}`,
       cu: `${SPECIAL_DELIM}header${SPECIAL_SEP}x-target-url${SPECIAL_DELIM}`
     }
-  }
+  },
+  weather_conditions: {
+    url: process.env["WEATHER_CONDITIONS_URL"],
+    query: {
+      apikey: process.env["WEATHER_KEY"],
```
Comment (htwyford) [inline server.js:57]: What's the process for getting these new environment variables added to the deployed server?

**Comment by sciurus** [comment]
@htwyford file a bug at https://bugzilla.mozilla.org/describecomponents.cgi?product=Cloud%20Services&component=Operations%3A%20Top%20Sites#Operations%3A%20Top%20Sites requesting them.

**Comment by htwyford** [comment]
Thanks. Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1683342

**Comment by mikedeboer** [comment]
@htwyford can you rebase your patch on top of the latest changes? I happened to land something that conflicts quite a bit with your changeset here!
I'll review asap when you did!

File: server.js [hunk 2]
```diff
@@ -48,56 +51,98 @@ const CONFIG = {
...
+    options.query.sub1 = "ebay";
+  }
+
+  if (options.url == process.env["WEATHER_CONDITIONS_URL"]) {
+    // The weather conditions API requires a dynamic key to be in the URL path
+    // rather than in the query paramters.
+    if (options.query.locationKey) {
+      options.url = `${options.url}${options.query.locationKey}.json`;
+    } else {
+      return;
```
Comment (mikedeboer) [inline server.js:109]: Why do this?

**Review by mikedeboer — CHANGES_REQUESTED** [review]

File: server.js (hunk 2 above)
Comment (htwyford) [reply server.js:109]: If the client is trying to access the weather conditions API but didn't provide a location key, we can't fetch any useful data. There's not much we can do but return.

Comment (mikedeboer) [reply server.js:109]: Alright, makes sense. Can you throw an error instead? We don't want the request to continue either in that case, right? Throwing an error will make the request fail with status code 500.

**Comment by htwyford** [comment]
Added an error thrown when the locationKey parameter is not provided to the weather_conditions endpoint.

**Review by mikedeboer — APPROVED** [review]

**Pull request merged**